File: SigreAppMobile/Hermes/common-ui/src/main/java/pe/com/hermes/common/util/LogHelper.java
package pe.com.hermes.common.util;

import android.util.Log;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;
import java.util.TimeZone;

/**
 * Logging con timestamp en hora de Lima y prefijo [HERMES] — equivalente
 * moderno de LogHelper.java de FastSales.
 */
public final class LogHelper {

    private static final SimpleDateFormat FORMATO = crearFormato();

    private LogHelper() {
    }

    private static SimpleDateFormat crearFormato() {
        SimpleDateFormat formato = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS", new Locale("es", "PE"));
        formato.setTimeZone(TimeZone.getTimeZone("America/Lima"));
        return formato;
    }

    private static String marca() {
        synchronized (FORMATO) {
            return "[HERMES " + FORMATO.format(new Date()) + "]";
        }
    }

    public static void i(String tag, String mensaje) { Log.i(tag, marca() + " " + mensaje); }
    public static void d(String tag, String mensaje) { Log.d(tag, marca() + " " + mensaje); }
    public static void w(String tag, String mensaje) { Log.w(tag, marca() + " " + mensaje); }
    public static void e(String tag, String mensaje) { Log.e(tag, marca() + " " + mensaje); }
    public static void e(String tag, String mensaje, Throwable error) { Log.e(tag, marca() + " " + mensaje, error); }
}

File: SigreAppMobile/Hermes/common-ui/src/main/java/pe/com/hermes/common/util/LogHelper.kt
package pe.com.hermes.common.util

import android.util.Log
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale
import java.util.TimeZone

/**
 * Logging con timestamp en hora de Lima y prefijo [HERMES] — equivalente
 * moderno de LogHelper.java de FastSales, útil para correlacionar logs de
 * dispositivo con logs de servidor (ambos en la misma zona horaria).
 */
object LogHelper {

    private val formato = SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS", Locale("es", "PE")).apply {
        timeZone = TimeZone.getTimeZone("America/Lima")
    }

    private fun marca(): String = "[HERMES ${formato.format(Date())}]"

    fun i(tag: String, mensaje: String) = Log.i(tag, "${marca()} $mensaje")
    fun d(tag: String, mensaje: String) = Log.d(tag, "${marca()} $mensaje")
    fun w(tag: String, mensaje: String) = Log.w(tag, "${marca()} $mensaje")
    fun e(tag: String, mensaje: String, error: Throwable? = null) = Log.e(tag, "${marca()} $mensaje", error)
}
